### PR TITLE
chore(FDS-523):[Table] Sort column based on the data type

### PR DESCRIPTION
**Pull request by rociobaena**
chore(FDS-523):[Table] Sort column based on the data type

## PR Author Checklist

- [x] PR title adheres to [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/#specification)
- [x] I have finished reviewing the contents of this PR for accuracy
- [x] If needed, I have included a [Changeset](https://github.com/changesets/changesets) and it follows [Semantic Versioning principles](https://semver.org/)
- [x] I have completed all of the above and have enabled `auto-merge` for this PR

### CODE REVIEW

**Review by odiseo42 — DISMISSED** [review]
LGTM

File: packages/cascara/src/components/Table/TableBase.js [hunk 1]
```diff
@@ -193,21 +183,22 @@ const TableBase = ({
...
+    const moduleType = display.find(
+      (item) => item.label === sortState?.sortAttribute
+    );
 
-    const sortedData = sortData(data);
+    const sortedData = sortDataType(
+      data,
+      sortState?.sortAttribute,
+      sortState?.sortOrder,
+      moduleType?.module
```
Comment (andresaraujo) [inline packages/cascara/src/components/Table/TableBase.js:195]: If `sortState` or `moduleType` are not present (null), will `sortDataType` function properly?

**Review by andresaraujo — DISMISSED** [review]

File: packages/cascara/src/components/Table/TableBase.js (hunk 1 above)
Comment (rociobaena) [reply packages/cascara/src/components/Table/TableBase.js:195]: Yes, if those properties don't exist, the sorting won't have any effect.

**Review by andresaraujo — APPROVED** [review]

**Comment by odiseo42** [comment]
testing

**Review by odiseo42 — APPROVED** [review]
LGTM

**Pull request merged**